tests/console/docker.pm

```diff
@@ -50,47 +50,62 @@ sub run {
     validate_script_output("docker search --no-trunc opensuse", sub { m/This project contains the stable releases of the openSUSE distribution/ });
 
     # images can be pulled from the Docker Hub
-    #   - pull minimalistic alpine image of declared version
+    #   - pull minimalistic alpine image of declared version using tag
     #   - https://store.docker.com/images/alpine
     my $alpine_image_version = '3.5';
     assert_script_run("docker pull alpine:$alpine_image_version", 300);
-    assert_script_run("docker images -q alpine:$alpine_image_version");
-    #   - pull hello-world image, typical docker demo image
+    assert_script_run("docker images alpine:$alpine_image_version | grep alpine");
```

Comment:
can you explain what this line suppose to prove ? Or do some real check or just drop it at all.

Reply:
@asmorodskyi,

This was the false negative.
```bash
sergio@sergio-latitude:~$ docker images none
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
sergio@sergio-latitude:~$ echo $?
0
```

So **assert_script_run** always passed before.

But grep comes handy here:

```bash
sergio@sergio-latitude:~$ docker images none | grep none
sergio@sergio-latitude:~$ echo $?
1
```

Would you prefer that I use **validate_script_output**?

----

drpaneas, the command before, was already filtering, no need to use `--filter referece`, `grep` is for ger rid of false negatives without having to extract the output or using validate output.

```bash
sergio@sergio-latitude:~$ docker images alpine:3.5
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
alpine              3.5                 6c6084ed97e5        3 weeks ago         3.99MB
```

Reply:
Interesting, I didn't know that docker filtering returns 0 in *any case*. This sounds like an upstream bug. Anyway, very good catch :D

Reply:
@SergioAtSUSE  now after I understand what you are trying to fix I am less skeptical , so you can leave at as it is , But I think validate_script_output would be better :)
@drpaneas interesting what docker guys would answer because for me it is not so clear that this is a bug - as similar example ls command query empty folder do you also think that it should return error if folder is empty ?

Reply:
@asmorodskyi, in this case you are filtering or searching for an image, so I also agree that it can be considered a bug.

Reply:
@asmorodskyi I don't like `validate_script_output` over `assert_script_run` because it involves more communication between the SUT and the worker and is not showing that clearly to a test reviewer what is the expectation

Reply:
in first case it get a string from SUT in another it get just exit code. Of course you can call this "involves more communication" but only if you have some hard feelings about `validate_script_output` :)

also not sure why you think that 
`assert_script_run("docker images alpine:$alpine_image_version | grep alpine");`
is more clear then 
validate_script_output("docker images alpine:$alpine_image_version", sub { m/alpine/});

Reply:
IMHO the weakest place of openQA is typing something. If there chance to reduce quantity of characters typed during test without loosing functionality I would vote for that without doubt

Reply:
but in both cases more is typed so you shouldn't just count the characters within perl code but what is visible in screenshots :)